pkg/scalers/aws_dynamodb_scaler.go

```diff
@@ -28,18 +25,18 @@ type awsDynamoDBScaler struct {
 }
 
 type awsDynamoDBMetadata struct {
-	tableName                 string
-	awsRegion                 string
-	awsEndpoint               string
-	keyConditionExpression    string
-	expressionAttributeNames  map[string]string
-	expressionAttributeValues map[string]types.AttributeValue
-	indexName                 string
-	targetValue               int64
-	activationTargetValue     int64
 	awsAuthorization          awsutils.AuthorizationMetadata
 	triggerIndex              int
 	metricName                string
+	TableName                 string                          `keda:"name=tableName, order=triggerMetadata"`
+	AwsRegion                 string                          `keda:"name=awsRegion, order=triggerMetadata"`
+	AwsEndpoint               string                          `keda:"name=awsEndpoint, order=triggerMetadata, optional"`
+	KeyConditionExpression    string                          `keda:"name=keyConditionExpression, order=triggerMetadata"`
+	ExpressionAttributeNames  map[string]string               `keda:"name=expressionAttributeNames, order=triggerMetadata"`
+	ExpressionAttributeValues map[string]types.AttributeValue `keda:"name=expressionAttributeValues, order=triggerMetadata"`
```

Comment:
@wozniakjan Do you have any advice on how to deal with this data type?
Currently, with this setup, keda is having these similar error when i run the unit test locally
```
=== RUN   TestParseDynamoMetadata/no_targetValue_given
    aws_dynamodb_scaler_test.go:333: 
        	Error Trace:	/home/tung/go/src/keda/pkg/scalers/aws_dynamodb_scaler_test.go:333
        	Error:      	Error "error parsing DynamoDb metadata: unable to set param \"expressionAttributeNames\" value \"{ \\\"#yr\\\" : \\\"year\\\" }\": expected format key=value, got \"{ \\\"#yr\\\" : \\\"year\\\" }\"\nunable to set param \"expressionAttributeValues\" value \"{\\\":yyyy\\\": {\\\"N\\\": \\\"1994\\\"}}\": expected format key=value, got \"{\\\":yyyy\\\": {\\\"N\\\": \\\"1994\\\"}}\"" does not contain "no targetValue given"
        	Test:       	TestParseDynamoMetadata/no_targetValue_given
```
I think it is expecting input value something like {yyyy=1994} instead :thinking:

Reply:
a path forward might be to add one more branch to the `TypedConfig` parsing logic for `types.AttributeValue`, similar to how `url.Values` are handled.

https://github.com/kedacore/keda/blob/5cbe424668646c00addb586ec4c84142f194fb6a/pkg/scalers/scalersconfig/typed_config.go#L373-L375

The logic for [`AttributeValue`](https://github.com/aws/aws-sdk-go-v2/blob/03768e0d0276b360a6abaa4d30318d4aedc44995/service/dynamodb/types/types.go#L54-L77) seems to be rather arbitrary to be handled purely by go native reflections package.

Reply:
Yeah I think I understand the issues:
- The separator now only supports `=`, while the input data uses `:` for kv separation. I will make a modification to support that. It will solve the issue for `ExpressionAttributeNames` https://github.com/kedacore/keda/blob/d4fcc84a8080e30923c40f6290a94f31804cb954/pkg/scalers/scalersconfig/typed_config.go#L66
- For `ExpressionAttributeValues`, how about we don't use `TypeConfig` but rely on the existing code. It is similar to `awsAuthorization` field https://github.com/kedacore/keda/blob/d4fcc84a8080e30923c40f6290a94f31804cb954/pkg/scalers/aws_dynamodb_scaler.go#L268

wdyt?

Reply:
> The separator now only supports `=`, while the input data uses `:`

it's worth trying but it seems like the input is a `json`? I guess for now both `ExpressionAttributeNames` and `ExpressionAttributeValues` might need to be excluded from `TypeConfig`, which is going to make schema generation a bit harder...

Reply:
Yes. Lets not touch it for now.